.travis.yml

```diff
@@ -8,51 +8,104 @@ addons:
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_grid
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_preview
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_configuration
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_keywords
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_ims
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_see_more
 cache:
   directories:
     - $HOME/.composer/cache
     - $HOME/.nvm
     - $HOME/node_modules
     - $HOME/yarn.lock
+    # cache location of webdrivers for MFTF (chromedriver, geckdriver, etc)
+    - $HOME/drivers
 before_install:
-  - mkdir -p ext/magento/adobe-stock-integration
-  - mv !(ext) ext/magento/adobe-stock-integration/
-  - ./ext/magento/adobe-stock-integration/dev/travis/before_install.sh
+  - bash -x dev/travis/before_install.sh
 install:
-  - cd magento2
+  - mkdir -p magento2/app/code/Magento
+  - mv Adobe* magento2/app/code/Magento/.
+  - pushd magento2
   - composer install
-  - composer config minimum-stability dev
-  - composer config repositories.ext path "../ext/*/*/*"
-  - composer require magento/adobe-stock-integration
+  - composer require astock/stock-api-libphp
+  - popd
 before_script:
-  - cp ../ext/magento/adobe-stock-integration/dev/tests/unit/phpunit.xml dev/tests/unit/phpunit.xml
-  - cp -f ../ext/magento/adobe-stock-integration/dev/travis/before_script.sh dev/travis/before_script.sh
-  - ./dev/travis/before_script.sh
+  - bash -x dev/travis/before_script.sh
 script:
-  - if [ $TEST_SUITE == 'unit' ]; then phpunit --configuration dev/tests/unit/phpunit.xml; fi
-  - if [ $TEST_SUITE == 'phpstan' ]; then composer require --dev phpstan/phpstan fooman/phpstan-magento2-magic-methods && phpstan analyse -l 2 ../ext -a dev/tests/api-functional/framework/autoload.php ; fi
-  - if [ $TEST_SUITE == 'static' ]; then phpcs --standard=dev/tests/static/framework/Magento/ ../ext ; fi
+  - pushd magento2
+  - if [ $TEST_SUITE == 'unit' ]; then vendor/bin/phpunit --configuration dev/tests/unit/phpunit.xml; fi
+  - if [ $TEST_SUITE == 'phpstan' ]; then composer require --dev phpstan/phpstan fooman/phpstan-magento2-magic-methods && vendor/bin/phpstan analyse -l 2 ../ext -a dev/tests/api-functional/framework/autoload.php ; fi
+  - if [ $TEST_SUITE == 'static' ]; then vendor/bin/phpcs --standard=dev/tests/static/framework/Magento/ ../ext ; fi
+  - if [ $TEST_SUITE == 'functional' ]; then echo "Running MFTF suite ${MFTF_SUITE}" && vendor/bin/mftf run:group $MFTF_SUITE --remove -vvv; fi
 after_success:
   - if [ $TEST_SUITE == 'unit' ]; then cd ../ && travis_retry coveralls; fi
+after_failure:
+  - if [ $TEST_SUITE == 'functional' ]; then cat "${HOME}/drivers/*.log"
```

Comment:
I think it would be good to also output `<magento>/var/log/exception.log`, `<magento>/var/log/system.log` and anything in `<magento>/var/reports/*` after failure